### PR TITLE
improve feeding performance

### PR DESCRIPTION
**Pull request by chobobobo**

### CODE REVIEW

File: room.js [hunk 1]
```diff
@@ -343,6 +343,15 @@ mod.extend = function(){
                     return this._urgentRepairableSites;
                 }
             },
+            'feedable': {
+                configurable: true,
+                get: function() {
+                    if (_.isUndefined(this._feedable)) {
+                        this._feedable = this.extensions.concat(this.spawns);
```
Comment (karlthepagan) [inline room.js]: the savings is probably from extensions and spawns using FIND_MY_STRUCTURES instead of FIND_STRUCTURES

Comment (chobobobo) [inline room.js]: They actually both load from memory caches which is the big savings.

**Comment by chobobobo** [comment]
Ah, this has a significant risk that new extensions and spawns will not be detected until the saveExtensions and saveSpawns is run again.. it requires more granular detection of structure changes.

**Comment by karlthepagan** [comment]
Let's save this for 0.8.0

Caching optimization makes me slightly reserved

**Comment by karlthepagan** [comment]
Changed my mind. We can put this in a minor revision but it should be tested separately.

**Pull request merged**